ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java

```diff
@@ -216,6 +216,7 @@ private void transferData(HttpURLConnection conn) {
                     errorOccurred = true;
                     updateDownloadFailed(DownloadManager.ERROR_FILE_ERROR,
                         "Error in creating destination file");
+                    return; // escape to finally to skip transferData()
```

Comment:
I have tested this fix to be working as expected.

Here is a [stack overflow](http://stackoverflow.com/questions/15225819/try-catch-finally-return-clarification)

Additionally, the [Oracle Java Finally Block Documentation](https://docs.oracle.com/javase/tutorial/essential/exceptions/finally.html) states that the finally block is intended to be used to ensure that a "return, continue, or break" statement doesn't skip cleanup code.